substrate/frame/staking-async/src/tests/unbonding_queue.rs

```diff
@@ -0,0 +1,603 @@
+fn get_min_lowest_stake_with_many_validators_works() {
+	ExtBuilder::default()
+		.validator_count(9)
+		.multi_page_election_provider(3)
+		// We already have the following validators:
+		// 11 -> 1000
+		// 21 -> 1000
+		.set_status(31, StakerStatus::<AccountId>::Validator) // 1000
+		.set_status(41, StakerStatus::<AccountId>::Validator) // 1000
+		.set_status(51, StakerStatus::<AccountId>::Validator) // 1000
+		.add_staker(61, 100, StakerStatus::<AccountId>::Validator)
+		.add_staker(71, 200, StakerStatus::<AccountId>::Validator)
+		.add_staker(81, 1000, StakerStatus::<AccountId>::Validator)
+		.add_staker(91, 1000, StakerStatus::<AccountId>::Validator)
+		// Add nominators
+		.set_status(101, StakerStatus::<AccountId>::Nominator(vec![11])) // 500
+		.add_staker(102, 500, StakerStatus::<AccountId>::Nominator(vec![21]))
+		.add_staker(103, 500, StakerStatus::<AccountId>::Nominator(vec![31]))
+		.add_staker(104, 500, StakerStatus::<AccountId>::Nominator(vec![41]))
+		.add_staker(105, 500, StakerStatus::<AccountId>::Nominator(vec![51]))
+		.add_staker(106, 500, StakerStatus::<AccountId>::Nominator(vec![61]))
+		.add_staker(107, 500, StakerStatus::<AccountId>::Nominator(vec![71]))
+		.add_staker(108, 500, StakerStatus::<AccountId>::Nominator(vec![81]))
+		.add_staker(109, 500, StakerStatus::<AccountId>::Nominator(vec![91]))
+		.build_and_execute(|| {
+			assert_ok!(Staking::set_staking_configs(
+				RuntimeOrigin::root(),
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Noop,
+				ConfigOp::Set(UnbondingQueueConfig {
+					min_slashable_share: Perbill::from_percent(50),
+					lowest_ratio: Perbill::from_percent(34),
+					unbond_period_lower_bound: 1,
+					back_of_unbonding_queue: Zero::zero(),
+				})
+			));
+
+			assert_eq!(Staking::get_min_lowest_stake(), 0);
+			Session::roll_until_active_era(2);
+
+			// There are 9 validators, so one third of them would be 3.
+			// Hence, the lowest third would be composed of the following validators:
+			// 61 -> 100 (own) + 500 (nomination from 106)
+			// 71 -> 200 (own) + 500 (nomination from 107)
+			// 31 -> 500 (own) + 500 (nomination from 103)
+			// Summing up 2300 total.
+			assert_eq!(Staking::get_min_lowest_stake(), 2300);
+		});
+}
+
+#[test]
+fn calculate_lowest_total_stake_works() {
+	ExtBuilder::default().has_stakers(false).build_and_execute(|| {
+		assert_ok!(Staking::set_staking_configs(
+			RuntimeOrigin::root(),
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Noop,
+			ConfigOp::Set(UnbondingQueueConfig {
+				min_slashable_share: Perbill::from_percent(50),
+				lowest_ratio: Perbill::from_percent(34),
+				unbond_period_lower_bound: 1,
+				back_of_unbonding_queue: Zero::zero(),
+			})
+		));
+
+		Session::roll_until_active_era(2);
+		Session::roll_until_active_era(3);
+		Session::roll_until_active_era(4);
```

Comment:
You can just do `roll_until_active_era(4)` and remove 2 and 3.

Reply:
Done in [1866f07](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/1866f076aa40cc6001a19fdad2b83be4a74ce359).